### DISCUSSION TITLE
How to checkpoint Datamodule's hyperparams/parameters on Model Checkpoint?

### DISCUSSION THREAD

**Discussion by FeryET** [Lightning-AI/pytorch-lightning#10172] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I want to create an inference model, and I have multiple datasets. For this issue, I want to save some settings about my DataModule and indicate what is the test indices, what is the dataset location and etc.

How can I add these to model_checkpoint?

**Comment by rohitgr7** [comment]
there is a `on_save_checkpoint` hook which receives the checkpoint (dict) before its saved. You can attach whatever you need inside your datamodule.

**Reply by FeryET** [reply]
@rohitgr7 

How to load the checkpoint info later into the datamodule? So far I use `LightningModule.load_from_checkpoint` to load a model, but I don't know how to parse the whole dict here?

**Reply by rohitgr7** [reply]
> parse the whole dict here?

can you elaborate more on this? `checkpoint` argument is a dict here. so you can easily access whatever you save inside `on_save_checkpoint` by `checkpoint[some_key]`.

**Reply by FeryET** [reply]
> > parse the whole dict here?
> 
> can you elaborate more on this? `checkpoint` argument is a dict here. so you can easily access whatever you save inside `on_save_checkpoint` by `checkpoint[some_key]`.

I have a LightningModule instance here, let's call it `PLModule` and a `LightningDataModule` let's call it `dm`. When I save a checkpoint via a trainer, `on_save_checkpoint` callbacks get called and something is saved in the modules. The problem is that when I want to load the checkpoint, `PLModule.load_checkpoint` clearly does not load the `LightningDataModule` information I want. I want to know if there is anyway to load the DataModule seperately without including it in the `PLModule` as a field?

**Reply by rohitgr7** [reply]
you mean `PLModule.load_from_checkpoint` here? if that's so then yes it won't load datamodule or callback related stuff since it just works when you resume your training. I think we should load datamodule states as well. Mind opening an issue? Let's see what others think :)
https://github.com/PyTorchLightning/pytorch-lightning/blob/dbe1662dc38d5217328ab459743b1113869a628c/pytorch_lightning/trainer/trainer.py#L1017-L1018
